hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/targettable/TargetTable.java

```diff
@@ -219,6 +220,12 @@ void onEvent(final TargetFilterEvent filterEvent) {
 
     @EventBusListenerMethod(scope = EventScope.UI)
     void onEvent(final ManagementUIEvent managementUIEvent) {
+      if(managementUIEvent == ManagementUIEvent.REFRESH_TARGETS_ON_FILTER_UPDATE)
```

Comment:
you need to first check if the table is filtered by the custom filter at all, otherwise you will unnecessarily refresh the table each time you modify some custom target filter elsewhere. I would also execute all of the logic in `UI.getCurrent().access` scope and combine tags conditionals with custom filter conditional (see above) and for both send the _ManagementUIEvent.TARGET_TABLE_FILTER_ event in order to update _CountMessageLabel_